lib/core/src/display/symbol.rs

```diff
@@ -258,8 +291,14 @@ impl Symbol {
             match uniform {
                 AnyUniform::Prim(uniform) =>
                     self.uniforms.push(UniformBinding::new(name,location,uniform)),
-                AnyUniform::Texture(uniform) =>
-                    self.textures.push(TextureBinding::new(name,location,uniform)),
+                AnyUniform::Texture(uniform) => {
+                    let texture_unit = texture_unit_iter.next().unwrap_or_else(|| {
+                        panic!("Texture unit limit exceeded.");
```

Comment:
If you can, do not panic! Emit error message and drop that texture. It is better to have a working program with glitched graphics than a crashed application.